docker-compose.yml

```
@@ -70,6 +70,23 @@ services:
       - SC_WEB_URL=http://web:8000
     depends_on:
       - web
+
+  py-sc-server:
+    build:
+      context: ./
+      dockerfile: ./problem-solver/py/Dockerfile
+    container_name: py-sc-server
+    restart: unless-stopped
+    depends_on:
+      problem-solver:
+        condition: service_healthy
+    environment:
+      - host=problem-solver
```

I don't think you have to configure it twice. 

Though you can try to read the whole url in the `server.py` script (akin to `SC_URL`), that would be more helpful because it would integrate into the existing `.env` of NIKA. 

You can also provide it with a default value and override it in case `SC_URL` was passed to `docker compose` (bash `${SOME_VAR:-default_value}` syntax, see [interpolation](https://docs.docker.com/compose/compose-file/12-interpolation/)).

It's a widespread convention to name env variables using BIG LETTERS and SNAKE_CASE, btw, so it would be cool if you'd change that too.
```suggestion
    environment:
      - "SC_URL=${SC_URL:-ws://problem-solver:8090/ws_json}"
```